universal-application-tool-0.0.1/app/services/applicant/question/AddressQuestion.java

```diff
@@ -200,6 +220,10 @@ private boolean isZipAnswered() {
    */
   @Override
   public boolean isAnswered() {
-    return isStreetAnswered() || isCityAnswered() || isStateAnswered() || isZipAnswered();
+    return isStreetAnswered()
+        || isLine2Answered()
```

Comment:
Question (non-blocking): Does this mean the line 2 is required? Usually it is optional, I think.

Reply:
+1, will make this blocking - I think Hung-Ming is right that it should be optional

Reply:
This does not make it required! I think this method might be kind of confusing... But basically, it's used to determine whether a question has been answered at all. Because we `||` (OR) all the `isAnswered` methods together, adding in `isLine2Answered()` doesn't actually effectively change the behavior here. All it does is say that if line 2 is answered at any point, this method will now evaluate to `true`, even if all other fields are not answered.

Reply:
Does that take away the requiredness of other parts viz street, city and zip?

Reply:
Thanks for the explanation! That does help a lot - looks good to me!

Reply:
Great question @AlaseI . `isAnswered` is not used to determine validation for required fields at all. I know this is sort of confusing. 😛 

`getAllTypeSpecificErrors` is the method that is used to check for those errors. `getAllTypeSpecificErrors` then checks for errors for each of the required fields. But notice that it doesn't check for line 2 to be answered.